### ISSUE TITLE
🥇v3 Release and feature plan

### ISSUE THREAD

**Issue by Its4Nik** [Its4Nik/DockStat-frontend#56]
# When will v3 be released?
Since the backend had some pretty huge changes it will take a longer time to fully develop a new UI / UX for the users.

# Tech Stack
- UI: Self built
- Framework: [Remix](https://remix.run)

# What can we expect from v2?
New features:
- Better DockStacks integration
- Container actions:
  - Start / Stop / Restart / Kill
  - Follow log lines
  - Remote Shell
- Better cup🥤 integration
- Better UX
- Main configuration (Themes / Hosts / ...) through the WebUI
- Public Status page for “public containers” => Useful for things like Minecraft Servers / Infrastructure.

# What are features that MIGHT be added?
## 1.
**DockStack-Agents**: This will be an Agent which will run on any server you'd like. The only purpose of this agent is to run stacks on servers where there is no "DockStatAPI". This might require a refactor of the current stack logic, or I'll just try to port it

## 2.
Some functionality like UptimeKuma's ping monitor, probably not as extensive as UptimeKuma, or Tianji and alike.

## 3.
An interactive Server Graph, with customization, it should be possible to export them as a link to embed them on your homepage to show off your amazing cluster 😎. The graph view can also be set as default view if it goes according to plan.

# How can I support the development?
Starring the repositories, helps me a lot!
https://github.com/its4nik/dockstat
https://github.com/its4nik/dockstatapi
https://github.com/its4nik/dockstacks

**Comment by Its4Nik** [comment]
# Update 23 Jan 2025

Hey there, this is the first development update in 2025!
There have been some rather big changes, please see below.

## Frontend development

I have finally decided on a Framework, its goind to be [Remix.run](https://remix.run) + [ShadCN](https://ui.shadcn.com/) + [TailwindCSS](https://tailwindcss.com/).
I am still figuring out a lot of quirks but keep an eye out for `v2.0.0-pre`

## Backend development

The backend has a MASSIVE change in how it works (compared to `v1`), new features:

### Features
 
- Stacks (Stacks are just an easy way to manage docker compose files from a webui)
- Connection Graphs (Needs some work done)
- High Availability
- Notification providers (Also need some work)
- And many more!

### Planned

I am going t admit it, the codebase of the Backend is not that clear, thats why a refactor is planned, probably in `v2.1`

## DockStacks

_Wait what is this?_

DockStacks is an "App Store" where you can get templates for (hopefully) all kind of container setups, currently there is a basic NGINX template and Outline Wiki template.

This feature will be available via the "App Store" inside the WebUI, or you can already visit it here: [DockStacks](https://its4nik.github.io/DockStacks)

## In Conclusion

This project is a lot of fun and I hope that someday the User base might grow :)
Thanks for coming to my TedTalk, see ya!

**Comment by Its4Nik** [comment]
# Update 4 Feb 2025

![Image](https://github.com/user-attachments/assets/ebbaa8bd-c416-412d-b190-0c5d4f390fd8)

![Image](https://github.com/user-attachments/assets/85a9bfaf-c85e-497b-9435-67b892c04603)

---

This update adds a bit more information to the Settings Overview page and a better error screen if the response from the API is empty / errored.

[[4cf8540](https://github.com/Its4Nik/DockStat/commit/4cf85408562678ba6d05cd20d6937204ca3161c9)] View the commit for more info

**Comment by Its4Nik** [comment]
# Update 7 Feb 2025

## Updated / Added

- Darkmode now works (with cookie)
- DockStacks is now integrated (only the store browsing functionalit)
- Sidebar Adjustments
- About Page added

## Screenshots

### DockStacks Store page
![Image](https://github.com/user-attachments/assets/212c5a99-6155-4285-931a-e474f0c51e66)

### DockStacks Page when opening a Stack
![Image](https://github.com/user-attachments/assets/7cd1abfd-19bb-4ad7-8f14-0098cfca7172)

### Settings Overview
![Image](https://github.com/user-attachments/assets/14314358-3d1c-4f29-9389-d42a36b2fa30)

### List view
![Image](https://github.com/user-attachments/assets/e1eba264-35f4-45f4-9ab0-68dcf7ea83ab)

### About Drawer
![Image](https://github.com/user-attachments/assets/4dd4bcf1-e7a6-45f3-aff2-e64ef5412418)

### Sidebar

|Theme|Collapsed|Expanded|
|-----|---------|--------|
|Dark|![Image](https://github.com/user-attachments/assets/cde59dc0-dd48-42e5-a553-993bb348023e)|![Image](https://github.com/user-attachments/assets/cf23f366-e25a-4acf-979f-898e2f78aef5)|
|Light|![Image](https://github.com/user-attachments/assets/729b1424-aca5-4a6a-bd8b-ff67594c1d45)|![Image](https://github.com/user-attachments/assets/3464f3fa-eeaa-48dd-b599-3545e0e85519)|
---

Commits:
- None

**Comment by Its4Nik** [comment]
# Update 9 Feb 2025

## Live logging:

![Image](https://github.com/user-attachments/assets/fb9713a0-2409-4520-8254-98f62fb723ba)

![Image](https://github.com/user-attachments/assets/623ebefb-dca0-4ac9-ab96-3bffe50d9290)

## Frontend Customisation:

![Image](https://github.com/user-attachments/assets/50b19024-7374-4ae3-bee8-53d7f821b13c)

![Image](https://github.com/user-attachments/assets/fdb4e411-9e0c-468a-9dd6-070363d4b085)

## Changelog:

- [`b922367`](https://github.com/Its4Nik/DockStat/commit/b922367ae2640e8d81b96e5f370000eeb5912959)

**Comment by Its4Nik** [comment]
# Update

![Image](https://github.com/user-attachments/assets/145f0086-c0cc-45de-99a1-db94e57cf981)

![Image](https://github.com/user-attachments/assets/ded59e22-7657-44eb-bec9-02d6d95d7f56)

New UI without library and way cooler theming!

**Comment by Its4Nik** [comment]
Uhmmm updates are on a halt, i have issues with bun crashing on windows

**Comment by Its4Nik** [comment]
# See #103 for the latest news!

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/160f5d05-8d82-4ebc-9759-980b1a758882" />
